doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+
+Setup
+=====
+
+To get started, you must satisfy the following prerequisites:
+
+* Specify an AWS Region where you'll train your model.
+* Create an IAM role ARN that is used to give Amazon SageMaker access to your data in Amazon Simple Storage Service (Amazon S3) necessary for training the model. See the `AWS IAM documentation <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html>`__ for how to fine tune the permissions needed.
+
+Capture real-time debugging data during model training in Amazon SageMaker
+==========================================================================
+
+To enable data emission for debugging model training, Amazon SageMaker will initialize a "hook" which, as the name suggests, would attach itself to the training process and emit the data necessary for debugging. To provide the hook's configuration, specify the new option called ``DebuggerHookConfig`` when training your model. For more about the ``DebuggerHookConfig`` object, see the `API documentation <https://sagemaker.readthedocs.io/en/stable/debugger.html#sagemaker.debugger.DebuggerHookConfig>`__.
+
+The ``DebuggerHookConfig`` accepts one or more objects of type ``CollectionConfig``. ``CollectionConfig`` defines the configuration around a tensor collection you intend to emit and save during model training. Collection object helps group tensors for easier handling.
+    as analysis
+
+.. code:: python
+
+    from sagemaker.debugger import CollectionConfig, DebuggerHookConfig
+
+    collection_config = CollectionConfig(
+        name='collection_name',
+        parameters={
+            "key": "value"
+        }
+    )
+
+    debugger_hook_config = DebuggerHookConfig(
+        s3_output_path='s3://path/for/data/emission',
+        container_local_output_path='/local/path/for/data/emission'
+        hook_parameters={
+            "key": "value"
+        },
+        collection_configs=[
+            collection_config,
+        ]
+    )
+
+    estimator = TensorFlow(
+        role=role
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        debugger_hook_config=debugger_hook_config
+    )
+
+When you call ``fit()`` on the ``estimator``, SageMaker creates a training job for you and the debugging hook is initialized using the parameters provided to the ``DebuggerHookConfig`` object. This hook starts emitting the relevant debugging data i.e. the tensor collections in real time and stores the data locally in the local path provided in ``DebuggerHookConfig``. This data is then uploaded to an S3 path derived from the path provided in the hook configuration in near real time:
```

Comment:
s/for you and the debugging hook is initialized/and initializes the debugging hook

also, add commas around "i.e. the tensor collections"

Reply:
* [@laurenyu ] Done. 
* [@aws-rbs ] Tried to reference SageMaker as Amazon SageMaker whenever used in isolation. Stuck with "SageMaker Debugger" and not "Amazon SageMaker Debugger" to avoid clunkiness. Let me know if that needs to be changed too.